File: ArvoreBinaria/Node.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ArvoreBinaria
{
    class Node
    {
        public string Valor { get; set; }
        public Node Esquerdo { get; set; }
        public Node Direito { get; set; }
        public Node Pai { get; set; }

        // Construtor principal, recebe o conteúdo e o pai do nó
        public Node(string valor, Node pai)
        {
            this.Valor = valor;
            this.Pai = pai;
            this.Direito = Esquerdo = null;
        }

        // Construtor para raiz
        public Node(string valor) : this(valor, null) { }
         
        public Boolean TemFilhoEsquerdo()
        {
            if (Esquerdo == null)
                return false;            
            return true;
        }

        public Boolean TemFilhoDireito()
        {
            if (Direito == null)
                return false;
            return true;
        }
     
        public override string ToString()
        {
            return this.Valor.ToString();
        }
    }
}

File: Prova_EDA/ArvoreBinaria/Node.py
class Node:
    def __init__(self, valor):
        self.valor = valor
        self.esquerdo = None
        self.direito = None
        self.pai = None

    def tem_filho_esquerdo(self) -> bool:
        return self.esquerdo is not None

    def tem_filho_direito(self) -> bool:
        return self.direito is not None

    def tem_2_filhos(self) -> bool:
        return self.temFilhoEsquerdo() and self.temFilhoDireito()

    def __str__(self):
        return str(self.valor)
